homeassistant/__init__.py

```diff
@@ -621,6 +621,16 @@ def set(self, entity_id, new_state, attributes=None):
         new_state = str(new_state)
         attributes = attributes or {}
 
+        # Last chance to enforce the visibility property. This is required for
+        # components that don't use the Entity base class for their entities.
+        # The sun component is an example of this. The Entity class cannot be
+        # imported cleanly, so assume the state is shown. This means that for
+        # visibility to be supported, the state must originate from a class
+        # that uses the base class Entity or it must manually put the hidden
+        # attribute in its attributes dictionary.
+        if ATTR_HIDDEN not in attributes:
```

Comment:
Yeah, that would make a lot more sense. It should be an easy update to the filter in the JavaScript.